modules/gke/main.tf

```diff
@@ -92,18 +18,16 @@ resource "google_container_cluster" "primary" {
   depends_on = [
     google_service_account.gke_sa,
     google_service_account.workload_identity_sa,
-    google_service_networking_connection.private_vpc_connection,
-    google_compute_subnetwork.subnet,
-    google_compute_route.default_route
+    var.network_dependency # This ensures the network is created first
```

Comment:
Shouldn't the gke module dependency on the networking dependency also ensure the network is created first?

I'm guessing you wouldn't have done this if it did, but that's weird.

Reply:
Good point, just tested this without the extra redundancy of the network_dependency var and it works well as well.